src/Entity/LineItem.php

```diff
@@ -93,84 +119,97 @@ public function getSlug(): string
+    public function disable(): self
     {
-        $this->startAt = $startAt;
+        $this->status = self::STATUS_DISABLED;
 
         return $this;
     }
 
+    public function getStartAt(): ?DateTimeInterface
+    {
+        return $this->startAt;
+    }
+
     public function getEndAt(): ?DateTimeInterface
     {
         return $this->endAt;
     }
 
-    public function setEndAt(?DateTimeInterface $endAt): self
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function setAvailabilityDates(DateTimeInterface $startAt = null, DateTimeInterface $endAt = null): self
     {
+        $this->validateAvailabilityDates($startAt, $endAt);
+
+        $this->startAt = $startAt;
         $this->endAt = $endAt;
 
         return $this;
     }
 
     public function isAvailableForDate(DateTimeInterface $date): bool
     {
-        if (null === $this->startAt || null === $this->endAt) {
-            return true;
+        if ($this->startAt !== null && $date < $this->startAt) {
+            return false;
         }
 
-        return $this->startAt <= $date && $this->endAt >= $date;
+        if ($this->endAt !== null && $date > $this->endAt) {
+            return false;
+        }
+
+        return true;
     }
 
     public function getMaxAttempts(): int
     {
         return $this->maxAttempts;
     }
 
-    public function setMaxAttempts(int $maxAttempts): self
-    {
-        $this->maxAttempts = $maxAttempts;
-
-        return $this;
-    }
-
     public function hasMaxAttempts(): bool
     {
         return $this->maxAttempts !== 0;
     }
 
-    public function isActive(): bool
-    {
-        return $this->isActive;
-    }
-
-    public function setIsActive(bool $isActive): self
-    {
-        $this->isActive = $isActive;
-
-        return $this;
-    }
-
     public function jsonSerialize(): array
     {
         return [
             'uri' => $this->getUri(),
             'label' => $this->getLabel(),
-            'isActive' => $this->isActive(),
-            'startDateTime' => $this->getStartAt() !== null ? $this->getStartAt()->getTimestamp() : '',
-            'endDateTime' => $this->getEndAt() !== null ? $this->getEndAt()->getTimestamp() : '',
+            'status' => $this->status,
+            'startDateTime' => $this->startAt !== null ? $this->startAt->getTimestamp() : '',
+            'endDateTime' => $this->endAt !== null ? $this->endAt->getTimestamp() : '',
             'maxAttempts' => $this->getMaxAttempts(),
         ];
     }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    private function validateAvailabilityDates(DateTimeInterface $startAt = null, DateTimeInterface $endAt = null): void
+    {
+        if (null !== $startAt && null !== $endAt && $startAt >= $endAt) {
+            throw new InvalidArgumentException(
+                "Invalid availability dates received. 'endAt' must be greater than 'startAt'."
+            );
+        }
+    }
```

Comment:
There is a problem here, it seems dates are not coming as null but:
 
```
class DateTime#463 (3) {
  public $date =>
  string(26) "1970-01-01 00:00:00.000000"
  public $timezone_type =>
  int(3)
  public $timezone =>
  string(3) "UTC"
}
```

This is causing problem during ingestion with null dates:

`[ERROR] Invalid availability dates received. 'endAt' must be greater than 'startAt'.`

Could you please have a look?

Reply:
In fact it seems to be coming from 

https://github.com/oat-sa/simple-roster/blob/feature/immutable-line-item/src/Command/Ingester/LineItemIngesterCommand.php#L174-L180

Reply:
Tried to ingest this:

```
uri,label,slug,startTimestamp,endTimestamp,maxAttempts
https://some_uri,"some_label",some_slug-QA,,,0
```

Reply:
Fixed, good catch